packages/main/src/plugin/statusbar/statusbar-registry.ts

```diff
@@ -51,6 +51,15 @@ export class StatusBarRegistry implements IDisposable {
     }
   }
 
+  // Change the icon of an entry and send an event to the renderer
+  changeEntryIcon(id: string, iconClass: string) {
+    const entry = this.entries.get(id);
+    if (entry) {
+      entry.entry.activeIconClass = iconClass;
+      this.apiSender.send(STATUS_BAR_UPDATED_EVENT_NAME, undefined);
+    }
+  }
+
```

Comment:
AFAIK we should not need that

entry belongs to someone, we can't change something that we don't own